lib/native/aws_iot.ts

```diff
@@ -368,8 +368,13 @@ export class AwsIotMqttConnectionConfigBuilder {
             "", username, authorizer_name, authorizer_signature, this.params.username);
         this.params.username = username_string;
         this.params.password = password;
-        this.tls_ctx_options.alpn_list = ["mqtt"];
-        this.params.port = 443;
+        if (this.params.use_websocket) {
+            // Do nothing - we can leave the port and alpn list as-is
+        }
+        else {
+            this.tls_ctx_options.alpn_list = ["mqtt"];
+            this.params.port = 443;
```

Comment:
Unclear why setting the port is conditional.  Custom auth requires port 443 no matter what.